pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -712,8 +712,8 @@ public void redeliverUnacknowledgedMessages() {
             });
             clearIncomingMessages();
             unAckedMessageTracker.clear();
+            resumeReceivingFromPausedConsumersIfNeeded();
```

Comment:
In `resumeReceivingFromPausedConsumersIfNeeded`:

```java
                internalPinnedExecutor.execute(() -> {
                    receiveMessageFromConsumer(consumer, true);
                });
```

Could you avoid calling `internalPinnedExecutor.execute` when `resumeReceivingFromPausedConsumersIfNeeded` is already called in `internalPinnedExecutor`?

Reply:
For multi-topic consumer, there is a lot of optimization to be done